streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -114,18 +115,28 @@ public StreamTask(TaskId id,
         // to corresponding source nodes in the processor topology
         partitionQueues = new HashMap<>();
 
-        TimestampExtractor timestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
+        final TimestampExtractor defaultTimestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
 
         for (TopicPartition partition : partitions) {
             SourceNode source = topology.source(partition.topic());
```

Comment:
When we add the sources by pattern (`KStreamBuilder.stream(final Pattern topicPattern)`), the source name is given like `"Pattern [" + regex + "]"`. For example, for `t.*` pattern it would be `"Pattern[t.*]"`. In `StreamTask`, we search for sources (in 121) by topic name. For example, `"topic1"`. However, the source name is `"Pattern[t.*]"`. So, it gives `null`.